Python/opcode_metadata.h

```diff
@@ -868,208 +875,208 @@ struct opcode_metadata {
+    [LOAD_FAST_LOAD_FAST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [LOAD_CONST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG | HAS_CONST_FLAG },
+    [STORE_FAST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [STORE_FAST_MAYBE_NULL] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [STORE_FAST_LOAD_FAST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [STORE_FAST_STORE_FAST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [POP_TOP] = { true, INSTR_FMT_IX, 0 },
+    [PUSH_NULL] = { true, INSTR_FMT_IX, 0 },
+    [END_FOR] = { true, INSTR_FMT_IB, 0 },
+    [INSTRUMENTED_END_FOR] = { true, INSTR_FMT_IX, 0 },
+    [END_SEND] = { true, INSTR_FMT_IX, 0 },
+    [INSTRUMENTED_END_SEND] = { true, INSTR_FMT_IX, 0 },
+    [UNARY_NEGATIVE] = { true, INSTR_FMT_IX, 0 },
+    [UNARY_NOT] = { true, INSTR_FMT_IX, 0 },
+    [UNARY_INVERT] = { true, INSTR_FMT_IX, 0 },
+    [BINARY_OP_MULTIPLY_INT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_ADD_INT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_SUBTRACT_INT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_MULTIPLY_FLOAT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_ADD_FLOAT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_SUBTRACT_FLOAT] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_ADD_UNICODE] = { true, INSTR_FMT_IBC, 0 },
+    [BINARY_OP_INPLACE_ADD_UNICODE] = { true, INSTR_FMT_IB, 0 },
+    [BINARY_SUBSCR] = { true, INSTR_FMT_IXC, 0 },
+    [BINARY_SLICE] = { true, INSTR_FMT_IX, 0 },
+    [STORE_SLICE] = { true, INSTR_FMT_IX, 0 },
+    [BINARY_SUBSCR_LIST_INT] = { true, INSTR_FMT_IXC, 0 },
+    [BINARY_SUBSCR_TUPLE_INT] = { true, INSTR_FMT_IXC, 0 },
+    [BINARY_SUBSCR_DICT] = { true, INSTR_FMT_IXC, 0 },
+    [BINARY_SUBSCR_GETITEM] = { true, INSTR_FMT_IXC, 0 },
+    [LIST_APPEND] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [SET_ADD] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [STORE_SUBSCR] = { true, INSTR_FMT_IXC, 0 },
+    [STORE_SUBSCR_LIST_INT] = { true, INSTR_FMT_IXC, 0 },
+    [STORE_SUBSCR_DICT] = { true, INSTR_FMT_IXC, 0 },
+    [DELETE_SUBSCR] = { true, INSTR_FMT_IX, 0 },
+    [CALL_INTRINSIC_1] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [CALL_INTRINSIC_2] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [RAISE_VARARGS] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [INTERPRETER_EXIT] = { true, INSTR_FMT_IX, 0 },
+    [RETURN_VALUE] = { true, INSTR_FMT_IX, 0 },
+    [INSTRUMENTED_RETURN_VALUE] = { true, INSTR_FMT_IX, 0 },
+    [RETURN_CONST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG | HAS_CONST_FLAG },
+    [INSTRUMENTED_RETURN_CONST] = { true, INSTR_FMT_IB, HAS_ARG_FLAG | HAS_CONST_FLAG },
+    [GET_AITER] = { true, INSTR_FMT_IX, 0 },
+    [GET_ANEXT] = { true, INSTR_FMT_IX, 0 },
+    [GET_AWAITABLE] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [SEND] = { true, INSTR_FMT_IBC, HAS_ARG_FLAG },
+    [SEND_GEN] = { true, INSTR_FMT_IBC, HAS_ARG_FLAG },
+    [INSTRUMENTED_YIELD_VALUE] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
+    [YIELD_VALUE] = { true, INSTR_FMT_IB, HAS_ARG_FLAG },
```

Comment:
I just noticed that this PR changed the format of `YIELD_VALUE/INSTRUMENTED_YIELD_VALUE` from `INSTR_FMT_IX` to `INSTR_FMT_IB`.  This is because we added the assertion that makes the generator identify it has HAS_ARG. I guess the new value is correct?

Reply:
In opcode.py it is classified as having an arg (by being >= HAVE_ARGUMENT) so I think it's correct. The oparg is used elsewhere to indicate the stack depth.